pyo3-macros-backend/src/module.rs

```diff
@@ -272,7 +272,7 @@ fn process_functions_in_module(options: &PyModuleOptions, func: &mut syn::ItemFn
                 let name = &func.sig.ident;
                 let statements: Vec<syn::Stmt> = syn::parse_quote! {
                     #wrapped_function
-                    #module_name.add_function(#krate::impl_::pyfunction::_wrap_pyfunction(&#name::DEF, #module_name)?)?;
+                    #module_name.add_function(#krate::impl_::pyfunction::_wrap_pyfunction(&#name::DEF, #module_name)?.into())?;
```

Comment:
@davidhewitt Did you have a plan for how to handle the new `_wrap_pyfunction_bound` here? I haven't worked out how to call the appropriate version of the function.

Reply:
I think you could use `wrap_pyfunction_bound` and use `#module_name.as_borrowed()` to force the module itself to always be the bound API.